multi-int-store/src/test/java/com/connexta/multiintstore/SolrTests.java

```diff
@@ -326,13 +335,63 @@ public String getFilename() {
   }
 
   @Test
-  @Ignore("TODO verify id matches something in S3 before storing to solr")
-  public void testStoreMetadataProductIdNotFound() {
-    // TODO verify 404? 400?
+  public void testStoreMetadataProductIdNotFound() throws Exception {
+    when(mockAmazonS3.doesObjectExist(anyString(), anyString())).thenReturn(Boolean.TRUE);
+
+    // given
+    final URI productLocation = new URI("http://localhost:8080/mis/product/12341");
+    final String queryKeyword = "Winterfell";
+    final String metadataContents =
+        "All the color had been leached from "
+            + queryKeyword
+            + " until only grey and white remained";
+    final String metadataEncoding = "UTF-8";
+    final InputStream metadataInputStream =
+        IOUtils.toInputStream(metadataContents, metadataEncoding);
+    final long metadataFileSize = (long) metadataInputStream.available();
+    final String metadataFileName = "test_file_name.txt";
+    final String metadataMimeType = "text/plain";
+    final MultiValueMap<String, Object> storeMetadataRequestBody = new LinkedMultiValueMap<>();
+    storeMetadataRequestBody.add("fileSize", String.valueOf(metadataFileSize));
+    storeMetadataRequestBody.add("mimeType", metadataMimeType);
+    storeMetadataRequestBody.add(
+        "file",
+        new InputStreamResource(metadataInputStream) {
+
+          @Override
+          public long contentLength() {
+            return metadataFileSize;
+          }
+
+          @Override
+          public String getFilename() {
+            return metadataFileName;
+          }
+        });
+    storeMetadataRequestBody.add("fileName", metadataFileName);
+    final HttpHeaders storeMetadataRequestHttpHeaders = new HttpHeaders();
```

Comment:
:bulb: It is probably time for our own HtttpHeaders factory. This might be something for a different PR, at your discretion.

I imagine it like:

```
@Bean
public class HttpHeadersFactory {

@Value("${acceptVersion}")
private final accept-version;

public static final String ACCEPT_VERSION = "Accept-Version"

public HttpHeaders createWithAcceptVersion() {
    headers = new HttpHeaders();
    headers.set(ACCEPT_VERSION,  accept-version);
    return headers;
}

public static HttpHeaders create() {
    return new HttpHeaders().createWithAcceptVersion();
}

```

Then line  381 becomes
```
  new HttpEntity<>(storeMetadataRequestBody, HttpHeadersFactory.create()));
```

We might even want an HttpEntityFactory that uses the HttpHeadersFactory. The client code would not know that an HttpHeadersFactory existed because it is encapsulated by a HttpEntityFactory

Reply:
I definitely think it should be a separate PR. I like the idea, but I'm not sure it's worth doing. Can you foresee a time when we have more than the custom `Accept-Version` header.